src/core.c/io_operators.pm6

```diff
@@ -125,10 +125,9 @@ multi sub close(Channel:D $channel) { $channel.close }
 
 proto sub slurp(|) {*}
 multi sub slurp(*%_) { $*ARGFILES.slurp(|%_) }
-multi sub slurp(IO::Handle:D $fh, *%_) { $fh.slurp(|%_) }
-multi sub slurp(IO() $path, :$bin!) { $path.slurp(:$bin) }
-multi sub slurp(IO() $path, :$enc ) { $path.slurp(:$enc) }
-multi sub slurp(IO() $path        ) { $path.slurp(:enc<utf8>) }
+multi sub slurp(IO::Handle:D $fh, *%_)    { $fh.slurp(|%_) }
+multi sub slurp(IO() $path, :$bin!)       { $path.slurp(:$bin) }
+multi sub slurp(IO() $path, :$enc='utf8') { $path.slurp(:$enc) }
```

Comment:
Why the reduction in candidates here?  The extra candidate exists to make dispatch faster.

Reply:
The only multi candidate that I cut is `multi sub slurp(IO() $path) { $path.slurp(:enc<utf8>) }`  (the diff shows three others as modified because I adjusted alignment).

I cut that one because, unless I'm confused, it's never called: `slurp('file')` dispatches to `slurp(IO() $path, $enc)` because it matches and is narrower.